homeassistant/components/samsungtv/__init__.py

```diff
@@ -1 +1,63 @@
 """The Samsung TV integration."""
+import socket
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+import homeassistant.helpers.config_validation as cv
+
+from .const import CONF_ON_ACTION, DEFAULT_NAME, DOMAIN
+
+
+def ensure_unique_hosts(value):
+    """Validate that all configs have a unique host."""
+    hosts = list()
+    for entry in value:
+        host = socket.gethostbyname(entry[CONF_HOST])
+        if host in hosts:
+            raise vol.Invalid("duplicate host entries found")
+        hosts.append(host)
+    return value
+
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.All(
+            cv.ensure_list,
+            ensure_unique_hosts,
```

Comment:
Maybe move this after the main schema so the host is validated as a string first?